lib/modules/programmingLanguages.nix

```diff
@@ -0,0 +1,37 @@
+{ nixpkgs, ... }@inputs:
+
+let
+  inherit (nixpkgs) lib;
+  inherit (lib)
+    mkEnableOption
+    mkOption
+    types
+    ;
+in
+rec {
+  programmingLanguagesModule = types.submodule ({ name, ... }: {
+    options = {
+      # enable = mkEnableOption "Enable ${name} support.";
+
+      vscode = mkOption {
+        type = vscodeModule;
+        default = { };
+      };
+    };
+  });
+
+  vscodeModule = with types; submodule {
```

Comment:
This should be generic with `tools`, but I think we had the same problem elsewhere, so let's leave it as-is for now.